ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,385 @@
+                if kind == "gamma":
+                    self.sim_info = sim_info
+                    self.spectrum = spectrum
+                bits = [header]
+                for start, stop, events in load.read_subarray_events_chunked(
+                    self.chunk_size
+                ):
+                    selected = self.epp.normalise_column_names(events)
+                    selected = selected[self.epp.get_table_mask(selected)]
+                    selected = self.make_derived_columns(
+                        kind, selected, spectrum, target_spectrum, obs_conf
+                    )
+                    bits.append(selected)
+
+                table = vstack(bits, join_type="exact")
+                reduced_events[kind] = table
+
+        select_ON = reduced_events["gamma"]["theta"] <= self.ON_radius * u.deg
+        self.signal = reduced_events["gamma"][select_ON]
+        self.background = vstack([reduced_events["proton"], reduced_events["electron"]])
+
+    def setup(self):
+        self.co = CutOptimising(parent=self)
+        self.e_bins = OutputEnergyBinning(parent=self)
+        self.bins = DataBinning(parent=self)
+        self.epp = EventPreProcessor(parent=self)
+
+        self.reco_energy_bins = self.e_bins.reco_energy_bins()
+        self.true_energy_bins = self.e_bins.true_energy_bins()
+        self.energy_migration_bins = self.e_bins.energy_migration_bins()
+
+        self.source_offset_bins = self.bins.source_offset_bins()
+        self.fov_offset_bins = self.bins.fov_offset_bins()
+        self.bkg_fov_offset_bins = self.bins.bkg_fov_offset_bins()
+
+    def start(self):
+        self.load_preselected_events()
+        self.gh_cuts, self.theta_cuts_opt, sens2 = self.co.optimise_gh_cut(
+            self.signal,
+            self.background,
+            self.alpha,
+            self.max_bg_radius,
+        )
+
+        self.signal["selected_theta"] = evaluate_binned_cut(
+            self.signal["theta"],
+            self.signal["reco_energy"],
+            self.theta_cuts_opt,
+            operator.le,
+        )
+        self.signal["selected"] = (
+            self.signal["selected_theta"] & self.signal["selected_gh"]
+        )
+
+        # calculate sensitivity
+        signal_hist = create_histogram_table(
+            self.signal[self.signal["selected"]], bins=self.reco_energy_bins
+        )
+
+        background_hist = estimate_background(
+            self.background[self.background["selected_gh"]],
+            reco_energy_bins=self.reco_energy_bins,
+            theta_cuts=self.theta_cuts_opt,
+            alpha=self.alpha,
+            fov_offset_min=self.bins.fov_offset_min * u.deg,
+            fov_offset_max=self.bins.fov_offset_max * u.deg,
+        )
+        self.sensitivity = calculate_sensitivity(
+            signal_hist, background_hist, alpha=self.alpha
+        )
+
+        # scale relative sensitivity by Crab flux to get the flux sensitivity
+        for s in (sens2, self.sensitivity):
```

Comment:
Not sure why you also scale `sens2`, which is never used.  What is the difference between self.sensitivity computed above, and sens2, which is computed during the optimization? Are they the same thing?

Reply:
This is a legacy of whatever script it I converted into the tool, I've just added saving sens2 to the output to check if there is any difference at all. For example, it might be that historically there was some difference but over time pyirf was refactored to use the same sensitivity function everywhere and now there is no difference.

Reply:
Which script are you referring to?

Reply:
The current example script in pyirf only computes sensitivity once using `optimize_gh_cuts`

Reply:
@maxnoe Probably this [protopipe](https://github.com/cta-observatory/protopipe/blob/master/protopipe/scripts/make_performance_EventDisplay.py) script. Is [this](https://github.com/cta-observatory/pyirf/blob/9c7aecc2e0491170adb83c55ddb54de4295714f8/examples/calculate_eventdisplay_irfs.py) the blessed script in pyIRF? The "Event display" in the name always makes me hesitant to conclude it is the appropriate script for `ctapipe` files.